File: machine-translation/microsoft-translator-text/MicrosoftTranslatorTextMachineTranslation.java
// "targetLanguages" is a list of strings. To translate into German and Italian, add "de" and "it" to that list.

public class TranslatorTextRequest {
    String Text;

    public TranslatorTextRequest(String text) {
        this.Text = text;
    }
}

public class LanguageDetection {
    public String language;
    public float score;
}

public class Translation {
    public String text;
    public String to;
}

public class TranslatorTextResponseItem {
    public LanguageDetection detectedLanguage;
    public List<Translation> translations;
}

public class TranslatorTextResponse extends ArrayList<TranslatorTextResponseItem> {}

String host = "https://api.cognitive.microsofttranslator.com";
String path = "/translate?api-version=3.0";
String base = host.concat(path);

URIBuilder builder = new URIBuilder(base);
for (String targetLanguage : targetLanguages) {
    builder.addParameter("to", targetLanguage);
}
String url = builder.toString();

TranslatorTextRequest request = new TranslatorTextRequest(input);
List<TranslatorTextRequest> requests = new ArrayList<>();
requests.add(request);

Gson gson = new Gson();
String body = gson.toJson(requests);

String responseString = Unirest.post(url)
    .header("Content-Type", "application/json")
    .header("Ocp-Apim-Subscription-Key", subscriptionKey)
    .body(body)
    .asString()
    .getBody();

TranslatorTextResponse response = gson.fromJson(responseString, TranslatorTextResponse.class);

List<Translation> translations = response.get(0).translations;

File: machine-translation/microsoft-translator-text/MicrosoftTranslatorTextMachineTranslation.kt
// "targetLanguages" is a list of strings. To translate into German and Italian, add "de" and "it" to that list.

data class TranslatorTextRequest(val Text: String)

data class LanguageDetection(val language: String, val score: Float)

data class Translation(val text: String, val to: String)

data class TranslatorTextResponseItem(
    val detectedLanguage: LanguageDetection,
    val translations: List<Translation>)

class TranslatorTextResponse : ArrayList<TranslatorTextResponseItem>()

val host = "https://api.cognitive.microsofttranslator.com"
val path = "/translate?api-version=3.0"
val base = host + path

val targetLanguages = listOf("de", "it")

val urlBuilder = URIBuilder(base)

targetLanguages.forEach { language ->
    urlBuilder.addParameter("to", language)
}

val url = urlBuilder
    .toString()

val request = TranslatorTextRequest(input)
val requests = ArrayList<TranslatorTextRequest>()
requests.add(request)

val gson = Gson()
val body = gson.toJson(requests)

val responseString = Unirest.post(url)
    .header("Content-Type", "application/json")
    .header("Ocp-Apim-Subscription-Key", subscriptionKey)
    .body(body)
    .asString()
    .body

val response = gson.fromJson(responseString, TranslatorTextResponse::class.java)

val translations = response[0].translations
